src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -28,89 +39,1027 @@ const ActorInit En_Js_InitVars = {
+            Animation_Change(&this->skelAnime, &gMoonChildGettingUpAnim, -1.0f,
+                             Animation_GetLastFrame(&gMoonChildGettingUpAnim) - 1.0f, 0.0f, 2, -10.0f);
+            this->unk_2B8 &= ~0x10;
+            this->unk_2B8 |= 8;
+        }
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    }
+    if (SkelAnime_Update(&this->skelAnime)) {
+        Animation_MorphToLoop(&this->skelAnime, &gMoonChildSittingAnim, -10.0f);
+        this->unk_2B8 &= ~8;
+    }
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFunc = func_8096A38C;
+        this->unk_2B8 &= ~2;
+        func_8096A184(this, play);
+        return;
+    }
+    if (!(this->unk_2B8 & 8) && (this->actor.xzDistToPlayer < 100.0f) &&
+        Player_IsFacingActor(&this->actor, 0x3000, play) && Actor_IsFacingPlayer(&this->actor, 0x1000)) {
+        func_800B8614(&this->actor, play, 120.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Js/func_8096A080.s")
+void EnJs_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnJs* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Js/func_8096A104.s")
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Js/func_8096A184.s")
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 40.0f, 25.0f, 40.0f, 5);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Js/func_8096A1E8.s")
+    this->actionFunc(this, play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Js/func_8096A2C0.s")
+    if ((this->cutsceneIndex != -1) && (ActorCutscene_GetCurrentIndex() != this->cutscenes[this->cutsceneIndex])) {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->cutscenes[this->cutsceneIndex]);
+        } else {
+            if (ActorCutscene_GetCanPlayNext(this->cutscenes[this->cutsceneIndex])) {
+                ActorCutscene_Start(this->cutscenes[this->cutsceneIndex], &this->actor);
+            } else {
+                ActorCutscene_SetIntentToPlay(this->cutscenes[this->cutsceneIndex]);
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Js/func_8096A38C.s")
+void func_8096A9F4(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    s32 pad;
+    EnJs* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Js/func_8096A6F4.s")
+    if (limbIndex == MOONCHILD_LIMB_HEAD) {
+        Matrix_MultVec3f(&D_8096AC30, &thisx->focus.pos);
+        if ((this->maskDlSelector >= 0) && (this->maskDlSelector < 5)) {
+            OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Js/EnJs_Update.s")
+            if (this->maskDlSelector != 0) {
```

Comment:
Personally I think `maskType` might be a better than `maskDlSelector`, along with an enum for each mask type 0 through 4 plus a MAX for 5.